### PR TITLE
Add espnetez

### PR DESCRIPTION
**Pull request by Masao-Someki**
## What?
This PR adds `espnetez` package to make it easier to use ESPnet!

## Why?
ESPnet runs primarily with a shell script, which can be difficult for beginners to use for running all the stages. The `espnetez` tool provides a Pythonic frontend for users, making it more user-friendly.

### CODE REVIEW

**Comment by Masao-Someki** [comment]
This is the sample training script for the ASR task with `espnetez`.
All you need is to build a dump file and pass it to the trainer.
You can easily join several datasets by just joining the dump file.

```python
import espnetez as ez

# dataset information
# the format of wav.scp is: <audio_tag><space><file_path>\n
# and the format of text is: <audio_tag><space><text>\n
# Example for wav.scp: audio_1 /database/libri100/train/first.flac
# Example for text: audio_1 HELLO WORLD
data_inputs = {
    "speech": { "file": "wav.scp", "type": "kaldi_ark" },
    "text":{ "file": "text", "type": "text" }
}
train_dump_path = "dump/raw/train_clean_100_sp"
test_dump_path = "dump/raw/test_clean"
output_path = "exp"

# You can use configuration from the ESPnet recipes.
training_config = ez.config.from_yaml("asr", "train_asr_branchformer_e24_amp.yaml")

# and you can update with your config.
preprocessor_config = ez.utils.load_yaml("preprocess.yaml")
training_config.update(preprocessor_config)

# Define trainer
trainer = ez.trainer.Trainer(
	"asr", train_dump_path, test_dump_path, output_path,
	data_inputs, training_config,
	ngpu=1 # you can also update configuration here
)

# If you don't have stats file then you need to run this.
trainer.collect_stats()

# finally run train()
trainer.train()

```

**Comment by Masao-Someki** [comment]
## ToDO
- [ ] Add more tasks. (Currently, I added asr/transducer/tts for debugging.)
- [ ] Add LM training
- [ ] Add sentencepiece training
- [ ] Add frontend for creating dump files

**Comment by sw005320** [comment]
@pyf98, can you help this project by reviewing and testing PRs?

**Comment by sw005320** [comment]
@Masao-Someki, can you let me know the progress?
Can you fix the CI and finish refactoring?
I think we can make an LM training step a lower priority.

**Comment by Masao-Someki** [comment]
@sw005320 
I'm using this espnetez to train a single ASR model but encountering some issues during training.
For example, the `uid` variable in the dataloader/dataset becomes `float` type, and I got assertion error.
I'm not sure why it happens, and debugging if the dump file is properly generated.

**Comment by Masao-Someki** [comment]
I added a demo notebook to train E-branchformer model with Librispeech-100 dataset. ([link](https://github.com/Masao-Someki/espnet/blob/feature/espnetez/egsez/asr/libri100.ipynb))
In my environment, the final `train()` function does not successfully executed on the Jupiter notebook. Maybe we need to run the `train()` function from command line.

**Comment by Masao-Someki** [comment]
The CI error says it failed to build some packages. Is there any available fix for this?

```
Failed to build grpcio future lxml distance
ERROR: Could not build wheels for grpcio, future, lxml, distance, which is required to install pyproject.toml-based projects
make: *** [espnet.done] Error 1
```

**Comment by sw005320** [comment]
Very cool!

@juice500ml, @ftshijt, @simpleoier, can you check this?

**Comment by pyf98** [comment]
Can we provide an example for fine-tuning OWSM (e.g., https://huggingface.co/espnet/owsm_v2_ebranchformer)? It will attract more users.

**Comment by juice500ml** [comment]
This PR is awesome!! Can we also consider packaging for pypi, so that people can easily `pip install espnetez` and directly use this?

**Comment by Masao-Someki** [comment]
Thank you @sw005320, @pyf98, and @juice500ml,

I apologize for the delay in development, but the bug in the fine-tuning process has been successfully fixed.
This PR is now ready for review!

Since I currently have only one GPU on my local machine, I kindly request the reviewer's assistance in checking whether the training process runs successfully with multiple GPUs.

File: espnet2/tasks/abs_task.py [hunk 1]
```diff
@@ -1064,7 +1064,6 @@ def print_config(cls, file=sys.stdout) -> None:
     @classmethod
     def main(cls, args: argparse.Namespace = None, cmd: Sequence[str] = None):
         assert check_argument_types()
-        print(get_commandline_args(), file=sys.stderr)
```
Comment (pyf98) [inline espnet2/tasks/abs_task.py:1067]: Why removing this?

**Comment by pyf98** [comment]
LGTM!

File: espnet2/tasks/abs_task.py (hunk 1 above)
Comment (Masao-Someki) [reply espnet2/tasks/abs_task.py]: @pyf98 
I'm sorry, I accidentally deleted this line...
I just reverted this modification.

**Comment by sw005320** [comment]
Thanks a lot, @Masao-Someki!
This is a great first step!

**Pull request merged**